src/main/java/com/laserfiche/repository/api/clients/impl/EntriesClientImpl.java

```diff
@@ -651,15 +656,20 @@ else if (rawResponse.getStatus() == 429)
                                         exception[0] = new RuntimeException(rawResponse.getStatusText());
                                 }
                             });
-                    if (exception[0] != null) {
-                        throw exception[0];
+                    shouldRetry = httpRequestHandler.afterSend(
+                            new ResponseImpl((short) statusCode[0])) || ApiClientUtils.isRetryableStatusCode(
+                            statusCode[0], HttpMethod.POST);
+                    if (!shouldRetry) {
+                        if (exception[0] != null) {
+                            throw exception[0];
+                        }
+                    } else {
+                        exception[0] = null;
                     }
                 } catch (Exception err) {
                     if (err instanceof ApiException) {
                         if (exception[0] != null) {
                             throw exception[0];
-                        } else {
-                            System.err.println(err);
                         }
                         break;
```

Comment:
a bit confused about the catch block. If the `err` is `ApiException` type but `exception[0]` is null then it will break the while loop and not throw any exception? Could this part be the same as the catch block in `ApiClientUtils.sendRequestWithRetry`? The new code is already throwing `exception[0]` if not retrying so do we still need the check for `exception[0] != null` in the catch?
```
            } catch (Exception err) {
                if (err instanceof ApiException || retryCount >= maxRetries) {
                    throw err;
                }
                shouldRetry = true;
            }
```

Reply:
yea i agree, going with the catch block logic in the `ApiClientUtils.sendRequestWithRetry` function would probably be the way to go, i think this might be something that i forgot to update (not too sure why i had it there initially)

Reply:
ok done